### PR TITLE
feat: 🎸 add endpoint workflow latest

### PR DESCRIPTION
**Pull request by dannil76**
## Explain the changes you’ve made
Added new endpoint `mypages/[hashid]/workflows/latest`

## Explain why these changes are made
To be able to check whether a case should be supplemented

## Explain your solution
Added new method - `get_workflow_latest` to the class VivaMyPages, who returns a case that is always the most recent

## How to test the changes?
1. check out branch
2. run `./server.sh`
3. Use Postman or alike and execute an GET request ie: `http://localhost:5000/mypages/[hasid]/workflows/latest`
You should receive one workflow object ie. the latest.

### CODE REVIEW

**Comment by sebastianthulin** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 7 Issues 
* Code issue 7 Code issues 

 [Details](https://app.embold.io/organization/gh/helsingborg-stad/repositories/0e85314a10805388831e336b2103fb2b/pull-request/49)

**Comment by helsingborgsstad** [comment]
Task linked: [CU-1pdvmgb Lägg till endpoint i VADA för att hämta ut senaste workflow](https://app.clickup.com/t/1pdvmgb)

File: app/libs/classes/viva_my_pages.py [hunk 1]
```diff
@@ -42,6 +42,13 @@ def get_workflow_list(self):
         except KeyError:
             raise Fault(message='No workflows found', code=404)
 
+    def get_workflow_latest(self):
+        try:
+            person_caseworkflow = self._get_person_caseworkflow(limit=1)
+            return person_caseworkflow['vivadata']['vivacaseworkflows']['workflow']
+        except KeyError:
+            raise Fault(message='No workflow latset found', code=404)
```
Comment (TeddyAndersson) [inline app/libs/classes/viva_my_pages.py:50]: This error message has incorrect english grammar.
Suggestion: "There was no workflow to be found" or "No workflow was found"

File: app/routes/my_pages_workflow_latest.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+class MyPagesWorkflowLatest(Resource):
+    method_decorators = [authenticate]
+
+    def get(self, hash_id):
+        personal_number = hash_to_personal_number(hash_id=hash_id)
+        self.my_pages = VivaMyPages(user=personal_number)
+
+        return self._get_workflow_latest()
+
+    def _get_workflow_latest(self):
```
Comment (TeddyAndersson) [inline app/routes/my_pages_workflow_latest.py:18]: This function name has incorrect english grammar
```suggestion
    def _get_latest_workflow(self):
```

File: app/routes/my_pages_workflow_latest.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class MyPagesWorkflowLatest(Resource):
+    method_decorators = [authenticate]
+
+    def get(self, hash_id):
+        personal_number = hash_to_personal_number(hash_id=hash_id)
+        self.my_pages = VivaMyPages(user=personal_number)
+
+        return self._get_workflow_latest()
```
Comment (TeddyAndersson) [inline app/routes/my_pages_workflow_latest.py:16]: Do we need to separate these functions ie get -> _get_workflow_latest, could we not simply cover everything in the get function? Then we would not need to concern about assigning values to `self`

Comment (TeddyAndersson) [reply app/routes/my_pages_workflow_latest.py:16]: Or is there a good reason for doing this ie flask-restful convention etc?

**Review by TeddyAndersson — CHANGES_REQUESTED** [review]
Find some minor changes, also missing tests. But not sure how to write test for flask-restful resources, would be good to start doing it though (y)

**Review by TeddyAndersson — APPROVED** [review]
Looks good ship it 🚀

**Pull request merged**